### PR TITLE
Add list of features and their status

### PR DESCRIPTION
**Pull request by sdake**
Resolves: #4142

First take at a feature list. The feature stage is not accurate,
but I think I have organized the various features into their various
dimensions.

Signed-off-by: Steven Dake <steven.dake@gmail.com>

### CODE REVIEW

**Comment by sdake** [comment]
To be clear, this review needs loads of help from the maintainers and especially the community that is relatively new to CHV to ensure correctness.

The feature state is definitely needed. If it would be easier, and I think it would, I could rework this PR to only be about features and we can address or not the feature state in the future or when I have learned far more about the project.

Ty

File: docs/features.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+### Image formats
+
+| Raw | Alpha |
+| Qcow2 | Alpha |
+| Fixed VHD | Alpha |
+| VHDX | Alpha |
+
+## Feature phase definitions
+
+| Stage | Description |
```
Comment (sboeuf) [inline docs/features.md:73]:
```suggestion
| Status | Description |
```

File: docs/features.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| --- | --- |
+| Experimental | The feature is in development, may not be finalized, and no support guarantees, including API guarantees, are provided. | 
+| Alpha | The feature is ready for evaluation, but the maintainers may remove the feature from a future release. | 
+| Beta |The feature is ready for production workloads, and some basic support guarantees are made, such as the feature will not be removed in the N-1 (e.g., last LTS release). | 
+| Stable | The feature is fialized, documented, tested both unit and functional, and guarantees are made about the API and its associated functionality. The maintainers will announce any feature deprecation with a timeline for removal, e.g., LTS+1 or LTS+2. |
```
Comment (sboeuf) [inline docs/features.md:78]:
```suggestion
| Stable | The feature is finalized, documented, tested both unit and functional, and guarantees are made about the API and its associated functionality. The maintainers will announce any feature deprecation with a timeline for removal, e.g., LTS+1 or LTS+2. | 
```

**Review by sboeuf — COMMENTED** [review]
- The description of the features looks good, but that's something everybody on the project should agree with.
- The rendering of the different tables is wrong. Try to find a way to make the markdown file to be rendered properly. Only the feature status definitions are rendered correctly into a table.
- You've used a lot of `Alpha` statuses, which I think should be `Beta` instead.
- I think we want to apply the title case for every single feature name, for consistency.

**Review by rbradford — COMMENTED** [review]
I'm not a fan of this approach. I think it is both too complicated and too simplistic. Take the recent deprecations and removals. They were specific options: dax for virtio-fs and mergeable for pmem that wouldn't fit into this approach.

I would prefer a list of:

a.) Experimental features which are subject to change without notice
b.) Pending removals (and thus active deprecations)

**Comment by rbradford** [comment]
Closing this as there's been no activity in > 21 days. Please feel free to reopen if necessary.

**Pull request closed without merging**